aws/resource_aws_wafv2_regex_pattern_set_test.go

```diff
@@ -0,0 +1,366 @@
+					resource.TestCheckResourceAttr(resourceName, "regular_expression_list.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAwsWafv2RegexPatternSet_Tags(t *testing.T) {
+	var v wafv2.RegexPatternSet
+	regexPatternSetName := fmt.Sprintf("regex-pattern-set-%s", acctest.RandString(5))
+	resourceName := "aws_wafv2_regex_pattern_set.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafv2RegexPatternSetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsWafv2RegexPatternSetConfig_OneTag(regexPatternSetName, "Tag1", "Value1"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafv2RegexPatternSetExists(resourceName, &v),
+					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "wafv2", regexp.MustCompile(`regional/regexpatternset/.+$`)),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Tag1", "Value1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccAWSWafv2RegexPatternSetImportStateIdFunc(resourceName),
+			},
+			{
+				Config: testAccAwsWafv2RegexPatternSetConfig_TwoTags(regexPatternSetName, "Tag1", "Value1Updated", "Tag2", "Value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafv2RegexPatternSetExists(resourceName, &v),
+					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "wafv2", regexp.MustCompile(`regional/regexpatternset/.+$`)),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Tag1", "Value1Updated"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Tag2", "Value2"),
+				),
+			},
+			{
+				Config: testAccAwsWafv2RegexPatternSetConfig_OneTag(regexPatternSetName, "Tag2", "Value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafv2RegexPatternSetExists(resourceName, &v),
+					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "wafv2", regexp.MustCompile(`regional/regexpatternset/.+$`)),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Tag2", "Value2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSWafv2RegexPatternSetDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_wafv2_regex_pattern_set" {
+			continue
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).wafv2conn
+		resp, err := conn.GetRegexPatternSet(
+			&wafv2.GetRegexPatternSetInput{
+				Id:    aws.String(rs.Primary.ID),
+				Name:  aws.String(rs.Primary.Attributes["name"]),
+				Scope: aws.String(rs.Primary.Attributes["scope"]),
+			})
+
+		if err == nil {
+			if *resp.RegexPatternSet.Id == rs.Primary.ID {
```

Comment:
we should guard against an unexpected `panic` here with a nil check before retrieving the `RegexPatternSet.Id` just in case :)

Reply:
@anGie44 you mean like this `aws.StringValue(resp.RegexPatternSet.Id) == rs.Primary.ID` or should `resp` and `resp.RegexPatternSet` also be checked?

Reply:
let's go with the former :) looking at our codebase, seems the pattern is to forgo the initial nil check of an object like `resp.RegexPatternSet`

Reply:
I don't mind adding the extra checks, but are these values not guaranteed to be non nil by the SDK in case of a successful operation?

Reply:
@pvanbuijtene unfortunately, the AWS Go SDK provides no guarantees and its up to the API remaining consistent. I would encourage adding `nil` checks where possible -- see https://github.com/terraform-providers/terraform-provider-aws/issues/3949 (and the plethora of crash reports around that time) for an example where missing `nil` checks caused issues from an API update.

Reply:
@bflad will keep that in mind for the future, thanks for the info 👍